wordpress-mobile/WordPress-Android
# Updated FluxC hash to avoid overwriting locally changed Posts

Fixes #8932 by bringing back the functionality first introduced in #3001 

The changes are in the corresponding FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1064

This PR makes sure to not overwrite posts that are marked as locally changed. Whenever the post list refreshes, local changes are retained.

This PR only brings back what was working before. Now we have revisions and are in a better position (in regards to the information we've got available) to implement something better. I've left further changes for other PRs to get a better UX (for example by expressly telling the user about the changes and such), waiting for design flow input in https://github.com/wordpress-mobile/WordPress-Android/issues/5984#issuecomment-452810170 - it may be good to go along the lines of what @oguzkocer suggested in https://github.com/wordpress-mobile/WordPress-Android/issues/5984#issuecomment-452836156

Also citing an obvious implication of this PR here (but important to note nevertheless), taken from #3001 description:

> 
> This does mean that if the user edits a post outside of the app, the app won't show those edits if it has local changes. This isn't ideal, but I think is still far better than the current situation.
> 

To test:
```
1. Create a draft while I am online in the app. That'll create the draft in remote.
2. Go offline
3. Make changes to the post and go back. This will save the changes as local draft
4. Go to web or any other client and make changes to the post.
5. Publish the post.
5.bis: now turn airplane mode off on the device and pull to refresh on the Posts list.
6. Observe the post, that was marked with "local changes" doesn't get updated and is still accessible.
```


Update release notes:
